### ISSUE TITLE
Problem: UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 95: ordinal not in range(128)

### ISSUE THREAD

**Issue by pA1nD** [s3tools/s3cmd#841]
Guys, I am running into an issue and can't figure out, how to fix.

Already fixed LANG and LC_ALL.
```
❯ locale
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
```

```
Invoked as: /usr/local/bin/s3cmd modify --recursive --add-header=Cache-Control:max-age=43200 s3://web-designio
Problem: UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 95: ordinal not in range(128)
S3cmd:   1.6.1
python:   2.7.13 (default, Feb 23 2017, 16:31:16) 
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
environment LANG=en_US.UTF-8

Traceback (most recent call last):
  File "/usr/local/bin/s3cmd", line 2919, in <module>
    rc = main()
  File "/usr/local/bin/s3cmd", line 2841, in main
    rc = cmd_func(args)
  File "/usr/local/bin/s3cmd", line 835, in cmd_modify
    return subcmd_cp_mv(args, s3.object_modify, "modify", u"modify: '%(src)s'")
  File "/usr/local/bin/s3cmd", line 815, in subcmd_cp_mv
    response = process_fce(src_uri, dst_uri, extra_headers)
  File "/usr/local/lib/python2.7/site-packages/S3/S3.py", line 805, in object_modify
    response = self.send_request(request)
  File "/usr/local/lib/python2.7/site-packages/S3/S3.py", line 1077, in send_request
    method_string, resource, headers = request.get_triplet()
  File "/usr/local/lib/python2.7/site-packages/S3/S3.py", line 199, in get_triplet
    self.sign()
  File "/usr/local/lib/python2.7/site-packages/S3/S3.py", line 195, in sign
    bucket_region, self.headers, self.body)
  File "/usr/local/lib/python2.7/site-packages/S3/Crypto.py", line 134, in sign_string_v4
    debug(u"canonical_headers = %s" % canonical_headers)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 95: ordinal not in range(128)
```

**Comment by fviard** [comment]
@pA1nD Can you try with the last "master" version of s3cmd? a lot of things were changed in this area to be more clean regarding unicode errors. 
Thanks

**Comment by fviard** [comment]
Should already be fixed in 2.0.0
